app/src/main/kotlin/com/simplemobiletools/calculator/helpers/BitwiseCalculatorImpl.kt

```diff
@@ -0,0 +1,302 @@
+                setValue(firstNumber.toString())
+                lastIsOperation = false
+            } else {
+                var value = firstNumber
+                firstNumber = convertBin(OCT, value)
+                setValue(firstNumber.toString())
+                lastIsOperation = false
+            }
+        } catch (e: NumberFormatException) {
+            setValue("OVERFLOW")
+        }
+    }
+
+    private fun convertDec(baseFrom: String, value: Long): Long {
+        if (baseFrom.equals(OCT)) {
+            if (value < 0) {
+                return (-1 * Integer.valueOf((value*-1).toString(), 8)).toLong()
+            } else {
+                return (Integer.valueOf(value.toString(), 8)).toLong()
+            }
+        } else if (baseFrom.equals(BIN)) {
+            if (value < 0) {
+                return (-1 * Integer.valueOf((value*-1).toString(), 2)).toLong()
+            } else {
+                return (Integer.valueOf(value.toString(), 2)).toLong()
+            }
+        } else return 0
+    }
+
+    private fun convertOct(baseFrom: String, value: Long): Long {
+        if (baseFrom.equals(DEC)) {
+            if (value < 0) {
+                return -1 * ((value * -1).toString(8)).toLong() // toOctalString with negative Long give errors
+            } else {
+                return (value.toString(8)).toLong()
+            }
+        } else if (baseFrom.equals(BIN)) {
+            var decimalValue = convertDec(BIN, value)
+            if (value < 0) {
+                return -1 * ((decimalValue * -1).toString(8)).toLong() // toOctalString with negative Long give errors
+            } else {
+                return decimalValue.toString(8).toLong()
+            }
+        } else return 0
+    }
+
+    private fun convertBin(baseFrom: String, value: Long): Long {
+        if (baseFrom.equals(DEC)) {
+            if (value < 0) {
+                return -1 * ((value * -1).toString(2)).toLong() // toBinaryString with negative Long give errors
+            } else {
+                return value.toString(2).toLong()
+            }
+        } else if (baseFrom.equals(OCT)) {
+            var decimalValue = convertDec(OCT, value)
+            if (value < 0) {
+                return -1 * ((decimalValue * -1).toString(2)).toLong() // toBinaryString with negative Long give errors
+            } else {
+                return decimalValue.toString(2).toLong()
+            }
+        } else return 0
+    }
+
+    private fun setValue(value: String) {
+        mCallback!!.setValue(value)
+    }
+
+    private fun setFormula(value: String) {
+        mCallback!!.setFormula(value)
+    }
+
+    private fun resetValues() {
+        firstNumber = 0
+        secondNumber = 0
+        operator = ""
+        secondNumberSet = false
+        setValue("0")
+        setFormula("")
+        digits = 0
+    }
+
+    fun addDigit(i: Long) {
+        if (operator != "") {
+            secondNumberSet = true
+            setClear()
+        }
+
```

Comment:
I am not 100% sure but I believe that's one too many spaces in here.

Reply:
Both ways are correct I believe in that case. I like to have my 'if' operations surrounded by single space unless the 'if' is followed by nothing but a closing bracket like a method ending or an 'if' inside an 'if', you will see:
    }
}

And never: 
    }

}

In that particular case, operations are done after the 'if' statement and I find it cleaner to add a space in between. The choice is yours, I can make a new commit for this single space!